src/bidiMapper/domains/context/browsingContextProcessor.ts

```diff
@@ -286,34 +289,62 @@ export class BrowsingContextProcessor {
   async process_script_addPreloadScript(
     params: Script.AddPreloadScriptParameters
   ): Promise<Script.AddPreloadScriptResult> {
-    const contexts: BrowsingContextImpl[] = [];
-    const scripts: Script.AddPreloadScriptResult[] = [];
-
-    if (params.context) {
-      // TODO(#293): Handle edge case with OOPiF. Whenever a frame is moved out
-      // of process, we have to add those scripts as well.
-      contexts.push(this.#browsingContextStorage.getContext(params.context));
-    } else {
-      // Add all contexts.
-      // TODO(#293): Add preload scripts to all new browsing contexts as well.
-      contexts.push(...this.#browsingContextStorage.getAllContexts());
-    }
+    const contexts: BrowsingContextImpl[] =
+      params.context === null
+        ? this.#browsingContextStorage.getAllContexts()
```

Comment:
For same-process frame, this will add the script twice, as it shares the CDP session, by which the CDP's `Page.addScriptToEvaluateOnNewDocument` is added.